core/vote/vote_manager.go

```diff
@@ -132,11 +135,22 @@ func (voteManager *VoteManager) loop() {
 
 			curHead := cHead.Block.Header()
 			// Check if cur validator is within the validatorSet at curHead
-			if !voteManager.engine.IsActiveValidatorAt(voteManager.chain, curHead) {
+			if !voteManager.engine.IsActiveValidatorAt(voteManager.chain, curHead,
+				func(bLSPublicKey *types.BLSPublicKey) bool {
+					return bytes.Equal(voteManager.signer.PubKey[:], bLSPublicKey[:])
+				}) {
 				log.Debug("cur validator is not within the validatorSet at curHead")
 				continue
 			}
 
+			// Add VoteKey to `miner-info`
+			once.Do(func() {
+				minerInfo := metrics.Get("miner-info")
+				if minerInfo != nil {
+					minerInfo.(metrics.Label).Value()["VoteKey"] = common.Bytes2Hex(voteManager.signer.PubKey[:])
+				}
```

Comment:
is it possible to move it into: `setMiner()`, cmd/utils/flags.go.

Reply:
tried and can't
before we set voteKey, we should wait two things:
1. the metrics.Label "miner-info" created
2. bls wallet loaded
here is the func called graph follow

makeFullNode
	makeConfigNode
		New
			NewVoteManager // here bls wallet loaded
		SetEthConfig
			setMiner
	SetupMetrics // here 'miner-info' created